src/Doctrine/DoctrineODMHelper.php

```diff
@@ -0,0 +1,352 @@
+                        $metadata[$m->getName()] = $m;
+                    }
+                }
+            }
+        }
+
+        return $metadata;
+    }
+
+    public function createDoctrineDetails(string $documentClassName): ?DocumentDetails
+    {
+        $metadata = $this->getMetadata($documentClassName);
+
+        if ($this->isInstanceOf($metadata, ClassMetadata::class)) {
+            return new DocumentDetails($metadata);
+        }
+
+        return null;
+    }
+
+    public function isClassAMappedDocument(string $className): bool
+    {
+        if (!$this->isDoctrineInstalled()) {
+            return false;
+        }
+
+        return (bool) $this->getMetadata($className);
+    }
+
+    /**
+     * Determines if the property-type will make the column type redundant.
+     *
+     * See ClassMetadataInfo::validateAndCompleteTypedFieldMapping()
+     */
+    public static function canColumnTypeBeInferredByPropertyType(string $columnType, string $propertyType): bool
+    {
+        return match ($propertyType) {
+            '\\'.\DateTime::class => Type::DATE === $columnType,
+            '\\'.\DateTimeImmutable::class => Type::DATE_IMMUTABLE === $columnType,
+            'array' => Type::HASH === $columnType,
+            'bool' => Type::BOOL === $columnType,
+            'float' => Type::FLOAT === $columnType,
+            'int' => Type::INT === $columnType,
+            'string' => Type::STRING === $columnType,
+            default => false,
+        };
+    }
+
+    public static function getPropertyTypeForColumn(string $columnType): ?string
+    {
+        return match ($columnType) {
+            Type::STRING => 'string',
+            Type::COLLECTION, Type::HASH => 'array',
+            Type::BOOL => 'bool',
+            Type::INT, Type::INTID, Type::INTEGER, Type::TIMESTAMP => 'int',
+            Type::FLOAT => 'float',
+            Type::DATE => '\\'.\DateTimeInterface::class,
+            Type::DATE_IMMUTABLE => '\\'.\DateTimeImmutable::class,
+            'uuid' => '\\'.Uuid::class,
+            'ulid' => '\\'.Ulid::class,
+            default => null,
+        };
+    }
+
+    /**
+     * Given the string "column type", this returns the "Type::STRING" constant.
+     *
+     * This is, effectively, a reverse lookup: given the final string, give us
+     * the constant to be used in the generated code.
+     */
+    public static function getTypeConstant(string $columnType): ?string
+    {
+        $reflection = new \ReflectionClass(Type::class);
+        $constants = array_flip($reflection->getConstants());
+
+        if (!isset($constants[$columnType])) {
+            return null;
+        }
+
+        return sprintf('Type::%s', $constants[$columnType]);
+    }
+
+    private function isInstanceOf($object, string $class): bool
+    {
+        if (!\is_object($object)) {
+            return false;
+        }
+
+        return $object instanceof $class;
+    }
```

Comment:
`$value instanceof $class` works even when `$value` is not an object. Not sure why you'd need this helper.

Reply:
DoctrineODMHelper is mostly based on DoctrineHelper as you noted. I have removed this method as minimum PHP version is 8.0 and `instanceof` acts exactly the same as the isInstanceOf method.
Am I supposed to make corresponding changes in DoctrineHelper in this PR?